### PR TITLE
Add Support for Ingress pathType and ingressClassName. Bump Puppetserver and PuppetDB Versions to Latest. Allow KubeVersion To Be Overridden.

### PR DESCRIPTION
**Pull request by Xtigyro**
+ Adding @underscorgan and @slconley as Code Owners too.

### CODE REVIEW

**Comment by Xtigyro** [comment]
@puppetlabs/platform-services
Could you please review the PR?

**Comment by Xtigyro** [comment]
@MikaelSmith @underscorgan @raphink 
Hey guys - could you please review the PR?

File: templates/puppetboard-ingress.yaml [hunk 1]
```diff
@@ -25,6 +27,9 @@ spec:
         paths:
           - path: /{{ rest $url | join "/" }}
             pathType: Prefix
```
Comment (MikaelSmith) [inline templates/puppetboard-ingress.yaml]: This seems odd if we're guarding pathType right after.

File: templates/puppetboard-ingress.yaml [hunk 2]
```diff
@@ -1,8 +1,10 @@
-{{- if .Values.puppetboard.enabled }}
-{{- if .Values.puppetboard.ingress.enabled }}
+{{- if and .Values.puppetboard.enabled .Values.puppetboard.ingress.enabled }}
 {{- $releaseName := .Release.Name -}}
 {{- $serviceName := "puppetdb" }}
 {{- $servicePort := .Values.puppetboard.port -}}
+{{- $pathType := .Values.puppetboard.ingress.pathType | default "ImplementationSpecific" -}}
+{{- $apiIsStable := eq (include "puppetserver.ingress.isStable" .) "true" -}}
+{{- $ingressSupportsPathType := eq (include "puppetserver.ingress.supportsPathType" .) "true" -}}
 apiVersion: networking.k8s.io/v1
```
Comment (MikaelSmith) [inline templates/puppetboard-ingress.yaml]: Shouldn't this use `puppetserver.ingress.apiVersion`?

File: templates/puppetboard-ingress.yaml [hunk 3]
```diff
@@ -1,8 +1,10 @@
-{{- if .Values.puppetboard.enabled }}
-{{- if .Values.puppetboard.ingress.enabled }}
+{{- if and .Values.puppetboard.enabled .Values.puppetboard.ingress.enabled }}
 {{- $releaseName := .Release.Name -}}
 {{- $serviceName := "puppetdb" }}
 {{- $servicePort := .Values.puppetboard.port -}}
+{{- $pathType := .Values.puppetboard.ingress.pathType | default "ImplementationSpecific" -}}
+{{- $apiIsStable := eq (include "puppetserver.ingress.isStable" .) "true" -}}
```
Comment (MikaelSmith) [inline templates/puppetboard-ingress.yaml:6]: Not clear what this is used for.

File: templates/puppetserver-ingress-compilers.yaml [hunk 4]
```diff
@@ -1,8 +1,10 @@
-{{- if .Values.puppetserver.compilers.enabled }}
-{{- if .Values.puppetserver.compilers.ingress.enabled -}}
+{{- if and .Values.puppetserver.compilers.enabled .Values.puppetserver.compilers.ingress.enabled -}}
 {{- $releaseName := .Release.Name -}}
 {{- $serviceName := include "puppetserver.puppetserver-compilers.serviceName" . }}
 {{- $servicePort := .Values.puppetserver.compilers.service.ports.puppetserver.port -}}
+{{- $pathType := .Values.puppetserver.compilers.ingress.pathType | default "ImplementationSpecific" -}}
+{{- $apiIsStable := eq (include "puppetserver.ingress.isStable" .) "true" -}}
```
Comment (MikaelSmith) [inline templates/puppetserver-ingress-compilers.yaml:6]: Not clear what this is used for.

File: templates/puppetserver-ingress-masters.yaml [hunk 5]
```diff
@@ -2,6 +2,9 @@
 {{- $releaseName := .Release.Name -}}
 {{- $serviceName := include "puppetserver.puppetserver-masters.serviceName" . }}
 {{- $servicePort := .Values.puppetserver.masters.service.ports.puppetserver.port -}}
+{{- $pathType := .Values.puppetserver.masters.ingress.pathType | default "ImplementationSpecific" -}}
+{{- $apiIsStable := eq (include "puppetserver.ingress.isStable" .) "true" -}}
```
Comment (MikaelSmith) [inline templates/puppetserver-ingress-masters.yaml:6]: Not clear what this is used for.

File: templates/puppetserver-ingress-compilers.yaml [hunk 6]
```diff
@@ -1,8 +1,10 @@
...
+{{- $ingressSupportsPathType := eq (include "puppetserver.ingress.supportsPathType" .) "true" -}}
 apiVersion: networking.k8s.io/v1
```
Comment (MikaelSmith) [inline templates/puppetserver-ingress-compilers.yaml]: Shouldn't this use `puppetserver.ingress.apiVersion`?

File: templates/puppetserver-ingress-masters.yaml [hunk 7]
```diff
@@ -2,6 +2,9 @@
...
+{{- $ingressSupportsPathType := eq (include "puppetserver.ingress.supportsPathType" .) "true" -}}
 apiVersion: networking.k8s.io/v1
```
Comment (MikaelSmith) [inline templates/puppetserver-ingress-masters.yaml]: Shouldn't this use `puppetserver.ingress.apiVersion`?

File: templates/puppetboard-ingress.yaml (hunk 1 above)
Comment (Xtigyro) [reply templates/puppetboard-ingress.yaml]: This is a leftover. Thanks.

Comment (Xtigyro) [reply templates/puppetboard-ingress.yaml]: Removed it.

File: templates/puppetboard-ingress.yaml (hunk 3 above)
Comment (Xtigyro) [reply templates/puppetboard-ingress.yaml:6]: Fixed - for optional ingress class name. Thanks.

File: templates/puppetboard-ingress.yaml (hunk 2 above)
Comment (Xtigyro) [reply templates/puppetboard-ingress.yaml]: Yup - fixed. Thanks.

File: templates/puppetserver-ingress-masters.yaml (hunk 5 above)
Comment (Xtigyro) [reply templates/puppetserver-ingress-masters.yaml:6]: Fixed. Thank you!

File: templates/puppetserver-ingress-compilers.yaml (hunk 6 above)
Comment (Xtigyro) [reply templates/puppetserver-ingress-compilers.yaml]: Not exactly - but tweaked it further. Thank you!

File: templates/puppetserver-ingress-masters.yaml (hunk 7 above)
Comment (Xtigyro) [reply templates/puppetserver-ingress-masters.yaml]: Not exactly - but tweaked it further. Thank you!

**Comment by Xtigyro** [comment]
@MikaelSmith All done - could you please review it once more? Thanks.

File: CODEOWNERS [hunk 8]
```diff
@@ -1,3 +1,3 @@
 # This repository is owned by Puppet's Platform Services team and community partners
 
-* @puppetlabs/platform-services @Xtigyro @raphink
+* @puppetlabs/platform-services @Xtigyro @underscorgan @slconley @raphink
```
Comment (MikaelSmith) [inline CODEOWNERS]: @underscorgan is already part of puppetlabs/platform-services

File: templates/puppetboard-ingress.yaml [hunk 9]
```diff
@@ -17,23 +19,34 @@ metadata:
...
             backend:
+              {{- if $apiIsStable }}
               service:
                 name: {{ $serviceName }}
                 port:
                   number: {{ $servicePort }}
+              {{- else }}
+              serviceName: {{ $serviceName }}
+              servicePort: {{ $servicePort }}
+    {{- end }}
```
Comment (MikaelSmith) [inline templates/puppetboard-ingress.yaml]: Indentation is off here, but probably doesn't indent actual behavior.

File: templates/puppetserver-ingress-compilers.yaml [hunk 10]
```diff
@@ -17,22 +19,34 @@ metadata:
...
             backend:
+              {{- if $apiIsStable }}
               service:
                 name: {{ $serviceName }}
                 port:
                   number: {{ $servicePort }}
+              {{- else }}
+              serviceName: {{ $serviceName }}
+              servicePort: {{ $servicePort }}
+    {{- end }}
```
Comment (MikaelSmith) [inline templates/puppetserver-ingress-compilers.yaml]: Same funny indentation here.

File: templates/puppetserver-ingress-masters.yaml [hunk 11]
```diff
@@ -16,18 +19,31 @@ metadata:
...
             backend:
+              {{- if $apiIsStable }}
               service:
                 name: {{ $serviceName }}
                 port:
                   number: {{ $servicePort }}
+              {{- else }}
+              serviceName: {{ $serviceName }}
+              servicePort: {{ $servicePort }}
+    {{- end }}
```
Comment (MikaelSmith) [inline templates/puppetserver-ingress-masters.yaml]: And here.

File: CODEOWNERS (hunk 8 above)
Comment (Xtigyro) [reply CODEOWNERS]: @MikaelSmith Fixed.

File: templates/puppetboard-ingress.yaml (hunk 9 above)
Comment (Xtigyro) [reply templates/puppetboard-ingress.yaml]: @MikaelSmith Fixed - though didn't affect anything.

Comment (Xtigyro) [reply templates/puppetboard-ingress.yaml]: @MikaelSmith Fixed - though didn't affect anything.

File: templates/puppetserver-ingress-masters.yaml (hunk 11 above)
Comment (Xtigyro) [reply templates/puppetserver-ingress-masters.yaml]: @MikaelSmith Fixed - though didn't affect anything.

File: templates/puppetserver-ingress-compilers.yaml (hunk 10 above)
Comment (Xtigyro) [reply templates/puppetserver-ingress-compilers.yaml]: @MikaelSmith Fixed - though didn't affect anything.

**Comment by Xtigyro** [comment]
@MikaelSmith All done - could you please review it once more? Thanks.

**Review by MikaelSmith — APPROVED** [review]

**Pull request merged**